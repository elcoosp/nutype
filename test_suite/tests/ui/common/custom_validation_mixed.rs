use nutype::nutype;

#[nutype(
    validate(with = validate_num, error = NumError, predicate = |val| *val > 0 )
)]
pub struct Num(i32);

fn validate_num(val: &i32) -> Result<(), NumError> {
    if *val > 100 {
        Err(NumError::TooBig)
    } else if *val < 0 {
        Err(NumError::TooSmall)
    } else {
        Ok(())
    }
}

#[derive(Debug)]
enum NumError {
    TooBig,
    TooSmall,
}

fn main () {}
