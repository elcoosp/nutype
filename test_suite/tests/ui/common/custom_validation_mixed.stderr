error: `with` and `error` attributes cannot be used mixed with other validators.
 --> tests/ui/common/custom_validation_mixed.rs:4:80
  |
4 |     validate(with = validate_num, error = NumError, predicate = |val| *val > 0 )
  |                                                                                ^
